src/components/TreeView.js

```diff
@@ -84,19 +85,15 @@ function RowValue(props) {
 
 function checkSignatures(signatures, fetchedSigners) {
   // catch fetch signature errors
-  if (fetchedSigners === null || fetchedSigners === "PENDING") {
-    signatures.state = SIGNATURE.NOT_VERIFIED_YET;
-    return
-  } else if (fetchedSigners === "ERROR") {
-    signatures.state = SIGNATURE.INVALID;
-    return
+  signatures.state = fetchedSigners.state;
```

Comment:
What's the purpose of this line? I don't see that `signatures` is referenced elsewhere.

I really try to avoid mutating data this way in React, especially in functions called in `render` because of how frequently (and unpredictably) they get run. You could end up in a situation where data gets changed multiple times, and it'd be really hard to figure out what's causing the data to change. Better to do handle all data mutations in the reducer.

Reply:
`signatures` is a reference to `node` (see call in [line 64](https://github.com/stellar/laboratory/pull/363/files/9f8e8e232e8baeb503eeb3830b52214d8143282c#diff-3fb8d04a71b32353f0ef387d0baf89d3R64)). It's not actually changing the state in React terminology, it's probably a name confusion: https://github.com/stellar/laboratory/pull/363#discussion_r253111030. We use `state` property too often, let's fix this in a separate PR.